banana24SourceBuild and centos8SourceBuild are broken in runtime rolling pipeline

I tracked it down to https://github.com/dotnet/runtime/pull/97725

Example build: https://dev.azure.com/dnceng-public/public/_build/results?buildId=551453&view=logs&j=d641c7e0-f8ca-569e-75ae-f8939bfb677f&t=ede234c6-4fb7-5936-436d-a8d770524e3b

```
dotnet-install: Installation finished successfully.

  Cloning repository at: /__w/1/s/ -> /__w/1/s/artifacts/sb/src/ ...
  Creating empty clone at: /__w/1/s/artifacts/sb/src/
  Cloning into '/__w/1/s/artifacts/sb/src'...
  done.
  Checking out sources...
  Clone complete: /__w/1/s/ -> /__w/1/s/artifacts/sb/src/
  Unsupported target OS 'centos.8'.
  Try 'build.sh --help' for values supported by '--os'.
/__w/1/s/.packages/microsoft.dotnet.arcade.sdk/9.0.0-beta.24081.4/tools/SourceBuild/SourceBuildArcadeBuild.targets(233,5): error MSB3073: The command "./build.sh  --arch x64 --os centos.8 --configuration Release --allconfigurations --verbosity minimal --nodereuse false --warnAsError false --outputrid centos.8-x64 /p:AdditionalRuntimeIdentifierParent= /p:ArcadeBuildFromSource=true /p:ContinuousIntegrationBuild=true /p:DotNetBuildInnerRepo=true /p:DotNetBuildSourceOnly=true /p:ArcadeInnerBuildFromSource=true /p:DotNetBuildFromSource=true /p:RepoRoot="/__w/1/s/artifacts/sb/src/" /p:ArtifactsDir="/__w/1/s/artifacts/sb/src/artifacts/" /bl:"/__w/1/s/artifacts/sb/src/artifacts/sourcebuild.binlog" /p:ContinuousIntegrationBuild=true /p:SourceBuildOutputDir="/__w/1/s/artifacts/sb/" /p:FullAssemblySigningSupported=false" exited with code 1. [/__w/1/s/.packages/microsoft.dotnet.arcade.sdk/9.0.0-beta.24081.4/tools/Build.proj]
##[error].packages/microsoft.dotnet.arcade.sdk/9.0.0-beta.24081.4/tools/SourceBuild/SourceBuildArcadeBuild.targets(233,5): error MSB3073: The command "./build.sh  --arch x64 --os centos.8 --configuration Release --allconfigurations --verbosity minimal --nodereuse false --warnAsError false --outputrid centos.8-x64 /p:AdditionalRuntimeIdentifierParent= /p:ArcadeBuildFromSource=true /p:ContinuousIntegrationBuild=true /p:DotNetBuildInnerRepo=true /p:DotNetBuildSourceOnly=true /p:ArcadeInnerBuildFromSource=true /p:DotNetBuildFromSource=true /p:RepoRoot="/__w/1/s/artifacts/sb/src/" /p:ArtifactsDir="/__w/1/s/artifacts/sb/src/artifacts/" /bl:"/__w/1/s/artifacts/sb/src/artifacts/sourcebuild.binlog" /p:ContinuousIntegrationBuild=true /p:SourceBuildOutputDir="/__w/1/s/artifacts/sb/" /p:FullAssemblySigningSupported=false" exited with code 1.

Build FAILED.

/__w/1/s/.packages/microsoft.dotnet.arcade.sdk/9.0.0-beta.24081.4/tools/SourceBuild/SourceBuildArcadeBuild.targets(233,5): error MSB3073: The command "./build.sh  --arch x64 --os centos.8 --configuration Release --allconfigurations --verbosity minimal --nodereuse false --warnAsError false --outputrid centos.8-x64 /p:AdditionalRuntimeIdentifierParent= /p:ArcadeBuildFromSource=true /p:ContinuousIntegrationBuild=true /p:DotNetBuildInnerRepo=true /p:DotNetBuildSourceOnly=true /p:ArcadeInnerBuildFromSource=true /p:DotNetBuildFromSource=true /p:RepoRoot="/__w/1/s/artifacts/sb/src/" /p:ArtifactsDir="/__w/1/s/artifacts/sb/src/artifacts/" /bl:"/__w/1/s/artifacts/sb/src/artifacts/sourcebuild.binlog" /p:ContinuousIntegrationBuild=true /p:SourceBuildOutputDir="/__w/1/s/artifacts/sb/" /p:FullAssemblySigningSupported=false" exited with code 1. [/__w/1/s/.packages/microsoft.dotnet.arcade.sdk/9.0.0-beta.24081.4/tools/Build.proj]
    0 Warning(s)
    1 Error(s)
```

--- a/eng/DotNetBuild.props
+++ b/eng/DotNetBuild.props
@@ -42,7 +42,7 @@
            This allows to build the repository using './build.sh <args> /p:DotNetBuildFromSource=true'.
            Properties that control flags from source-build, and the expected output for source-build should be added to this file. -->
       <InnerBuildArgs>$(InnerBuildArgs) $(FlagParameterPrefix)arch $(TargetArch)</InnerBuildArgs>
-      <InnerBuildArgs>$(InnerBuildArgs) $(FlagParameterPrefix)os $(TargetOS)</InnerBuildArgs>
+      <InnerBuildArgs Condition="'$(ShortStack)' == 'true'">$(InnerBuildArgs) $(FlagParameterPrefix)os $(TargetOS)</InnerBuildArgs>
       <InnerBuildArgs Condition="'$(TargetArch)' != '$(_hostArch)' and '$(ShortStack)' != 'true'">$(InnerBuildArgs) $(FlagParameterPrefix)cross</InnerBuildArgs>
       <InnerBuildArgs>$(InnerBuildArgs) $(FlagParameterPrefix)configuration $(Configuration)</InnerBuildArgs>
       <InnerBuildArgs Condition="'$(ShortStack)' != 'true'">$(InnerBuildArgs) $(FlagParameterPrefix)allconfigurations</InnerBuildArgs>

--- a/eng/DotNetBuild.props
+++ b/eng/DotNetBuild.props
@@ -42,7 +42,7 @@
            This allows to build the repository using './build.sh <args> /p:DotNetBuildFromSource=true'.
            Properties that control flags from source-build, and the expected output for source-build should be added to this file. -->
       <InnerBuildArgs>$(InnerBuildArgs) $(FlagParameterPrefix)arch $(TargetArch)</InnerBuildArgs>
-      <InnerBuildArgs Condition="'$(ShortStack)' == 'true'">$(InnerBuildArgs) $(FlagParameterPrefix)os $(TargetOS)</InnerBuildArgs>
+      <InnerBuildArgs Condition="'$(DotNetBuildSourceOnly)' != 'true'">$(InnerBuildArgs) $(FlagParameterPrefix)os $(TargetOS)</InnerBuildArgs>
       <InnerBuildArgs Condition="'$(TargetArch)' != '$(_hostArch)' and '$(ShortStack)' != 'true'">$(InnerBuildArgs) $(FlagParameterPrefix)cross</InnerBuildArgs>
       <InnerBuildArgs>$(InnerBuildArgs) $(FlagParameterPrefix)configuration $(Configuration)</InnerBuildArgs>
       <InnerBuildArgs Condition="'$(ShortStack)' != 'true'">$(InnerBuildArgs) $(FlagParameterPrefix)allconfigurations</InnerBuildArgs>

